src/Converter/ShippingAdjustmentsToLineItemsConverter.php

```diff
@@ -59,13 +61,15 @@ private function convertShippingAdjustmentToLineItem(AdjustmentInterface $shippi
         $grossValue = $shipment->getAdjustmentsTotal();
         $taxAdjustment = $this->getShipmentTaxAdjustment($shipment);
         $taxAmount = $taxAdjustment !== null ? $taxAdjustment->getAmount() : 0;
-        $netValue = $grossValue - $taxAmount;
+        $unitNetPrice = $grossValue - $taxAmount - $shipment->getAdjustmentsTotal(AdjustmentInterface::ORDER_SHIPPING_PROMOTION_ADJUSTMENT);
```

Comment:
I have the feeling, that it is small de-cohesion, that in the units to line items converter we have dedicated service for net price calculation and here we don't

Reply:
Indeed 👍 we could probably refactor (could be done in a separate PR IMO) 🖖